### DISCUSSION TITLE
How to bypass the "Do you want to change your default shell?" question

### DISCUSSION THREAD

**Discussion by agusmakmun** [ohmyzsh/ohmyzsh#12796] [Help] [answered]
I want to create automation script to setup the laptop, and one of them is to install this ohmyzsh;

```bash
apt install zsh
sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
```

but it's prompting to use confirm whether we want to use ohmyzsh as default shell or not:

```
Time to change your default shell to zsh:
Do you want to change your default shell to zsh? [Y/n] y
Changing your shell to /usr/bin/zsh...
Shell successfully changed to '/usr/bin/zsh'.
```

How to make it default `yes`?

Using this `-y` is not working;

```bash
sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" -y
```

and using this will not switch to default;

```bash
sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" "" --unattended

# or

echo "y" | sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
```

**Comment by vraravam** [comment]
After installing the zsh, use this command: `chsh -s $(which zsh)`

For more info, look [here](https://www.howtogeek.com/669835/how-to-change-your-default-shell-on-linux-with-chsh/)

**Comment by carlosala** [comment, marked as answer]
Passing `--skip-chsh` will do the trick. [Here](https://github.com/ohmyzsh/ohmyzsh/blob/7ed475cb589c9e82211f71b3a5d7083b69cea93c/tools/install.sh#L3-L39) you have the docs of the installer.
Thanks!
